openreview/arr/helpers.py

```diff
@@ -1481,106 +1480,135 @@ def flag_submission(
+    ))
+    ethics_flag_from_reviews = False
+    dsv_flag_from_reviews = False
+    for review in reviews:
+        # Check for ethics flagging
+        print(f"ethics review flag state {ethics_flag_from_reviews}")
+        ethics_flag_from_reviews = ethics_flag_from_reviews or check_field_violated(
+            review,
+            ethics_flag_fields['Review'],
+            ethics_flag_default
+        )
+        # Check for desk reject verification
+        for violation_field, field_default in violation_fields['Official_Review'].items():
+            print(f"dsv review flag state {dsv_flag_from_reviews}")
+            dsv_flag_from_reviews = dsv_flag_from_reviews or check_field_violated(
+                review,
+                violation_field,
+                field_default
             )
-                
 
-    # Desk Rejection Flagging
-    print('checking for dsv')
-    if not all(check_field_not_violated(edit.note.content, field) for field in violation_fields) and not dsv_flagged:
-        print('flagging dsv')
-        post_flag(
-           'Desk_Reject_Verification',
-           value = True
+    # Check checklists
+    checklists = list(filter(
+        lambda reply: any('Checklist' in inv for inv in reply['invitations']),
+        forum.details['replies']
+    ))
+    ethics_flag_from_checklists = False
+    dsv_flag_from_checklists = False
+    for checklist in checklists:
+        # Check for ethics flagging
+        print(f"ethics checklist flag state {ethics_flag_from_checklists}")
+        ethics_flag_from_checklists = ethics_flag_from_checklists or check_field_violated(
+            checklist,
+            ethics_flag_fields['Checklist'],
+            ethics_flag_default
         )
-    else:
-        # Check for unflagging
-        checklists = list(filter(
-           lambda reply: any(reply_name in inv for inv in reply['invitations']),
-           forum.details['replies']
-        ))
-
-        print(f"{len(checklists)} valid responses for unflagging")
-
-        dsv_unflag = True
-        for checklist in checklists:
-            dsv_unflag = dsv_unflag and all(check_field_not_violated(checklist['content'], field) for field in violation_fields)
-
-        if dsv_unflag and has_dsv_flag_history and dsv_flagged:
-            post_flag(
-                'Desk_Reject_Verification',
-                value = False
+        # Check for desk reject verification
+        for violation_field, field_default in violation_fields['Checklist'].items():
+            print(f"dsv flag state {dsv_flag_from_checklists}")
+            dsv_flag_from_checklists = dsv_flag_from_checklists or check_field_violated(
+                checklist,
+                violation_field,
+                field_default
             )
-    
-    # Ethics Flagging
-    if needs_ethics_review and not has_ethic_flag_history:
-        print('flagging ethics and emailing')
+
+    # Check ethics flag
+    ## False -> True
+    if not ethics_flagged and any([
```

Comment:
Instead of getting `ethics_flagged` at the beginning of the process function. I would get it at the moment that we evaluate if we need to flag it or not to avoid race conditions.